src/lexer.coffee

```diff
@@ -889,7 +889,11 @@ exports.Lexer = class Lexer
           converted = fn.call this, token[1], i
           addTokenData token, initialChunk: yes if i is 0
           addTokenData token, finalChunk: yes   if i is $
-          addTokenData token, {indent, quote, double}
+          addTokenData token, {
+            indent
+            quote: quote unless quote is '///'
```

Comment:
After updating the containing `StringWithInterpolations` of a heregex to have `quote: '///'` (see above comment), the string-chunks (`StringLiteral`s) inside of it were deciding that their quote was `/` (since they do something like `@quote = quote.charAt 0`)

This was causing test breakage and doesn't make sense, so don't "pass through" `quote` when it's `///`

Reply:
Wouldn’t `quote` be inconsistent then? Why not update `@quote =` to be something like
```coffee
@quote = if quote.startsWith('///') then `'///'` else quote.charAt 0
```

Reply:
Ok to clarify, there's two different places where `quote` is being applied here:

On the "outer" `StringWithInterpolations` (corresponding to the whole interpolated heregex), it's now getting passed `quote: '///'` (whereas before it didn't get passed `quote` at all)

Then on the inner `StringLiteral`s representing "text chunks" inside the interpolated heregex, `quote` really doesn't apply at all (this is the same idea that I was discussing in the previous `StringWithInterpolations` PR, that actual string literals with quotes are sort of different from text-chunks inside of an interpolated string). With the "outer" change to `quote: '///'`, this "inner" was also updating to `quote: '///'` since here the existing logic was for them both to use the same `quote`

So I think this is the cleanest place to specify that "when we're dealing with text chunks inside an interpolated heregex, make sure that they don't decide that their quote is `'///'`/`'/'`" (and then in `StringLiteral::constructor()`, `quote` will default to `'"'`)

Reply:
Right but isn’t it weird that the other similar tokens have a `quote` property and this one doesn’t? Shouldn’t they be consistent, and we handle all the various quote possibilities (both 1-character and 3-character) on the other side?

Reply:
Sure, removed the special case here and shifted into `nodes.coffee`